### PR TITLE
Fix BROOKLYN-249

### PR DESCRIPTION
**Pull request by aledsage**
See more detailed descriptions in each commit message.

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/location/byon/ByonLocationResolverRebindTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.brooklyn.api.location.LocationSpec;
+import org.apache.brooklyn.api.location.MachineLocation;
+import org.apache.brooklyn.api.location.MachineProvisioningLocation;
+import org.apache.brooklyn.core.mgmt.rebind.RebindTestFixtureWithApp;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableMap;
+
+public class ByonLocationResolverRebindTest extends RebindTestFixtureWithApp {
```
Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/location/byon/ByonLocationResolverRebindTest.java]: This should have been included in https://github.com/apache/brooklyn-server/pull/91, but was accidentally missed out!

Not quite appropriate to include it in this commit, but no big deal in my opinion.

**Comment by neykov** [comment]
Looks good. Are the external properties in locations already covered by tests (don't see related changes in last commit)?

**Comment by aledsage** [comment]
Existing tests include `ExternalConfigYamlTest.testExternalisedLocationConfigReferencedFromYaml` and `testExternalisedLocationConfigSetViaProvisioningPropertiesReferencedFromYaml`.

We could do with more tests for other specific types of location. For example, I worry that a BYON won't correctly resolve external config.

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/JcloudsRebindWithExternalConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
+package org.apache.brooklyn.camp.brooklyn;
```
Comment (sjcorbett) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/JcloudsRebindWithExternalConfigTest.java]: This is missing the ASF license header.

Comment (sjcorbett) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/JcloudsRebindWithExternalConfigTest.java]: I'll just push a fix to master.

**Pull request merged**